solutions/project/project-sandbox/network/subnet.yaml

```diff
@@ -0,0 +1,158 @@
+  privateIpGoogleAccess: true
+  networkRef:
+    name: project-id-global-vpc1-vpc # kpt-set: ${project-id}-global-vpc1-vpc
+  logConfig:
+    aggregationInterval: INTERVAL_5_SEC
+    flowSampling: 0.5
+    metadata: INCLUDE_ALL_METADATA
+---
+# Subnet APPRZ northamerica-northeast1
+# SC-7 BOUNDARY PROTECTION
+apiVersion: compute.cnrm.cloud.google.com/v1beta1
+kind: ComputeSubnetwork
+metadata:
+  name: project-id-nane1-vpc1-apprz-snet # kpt-set: ${project-id}-nane1-vpc1-apprz-snet
+  namespace: networking
+  annotations:
+    cnrm.cloud.google.com/project-id: project-id # kpt-set: ${project-id}
+    config.kubernetes.io/depends-on: compute.cnrm.cloud.google.com/namespaces/networking/ComputeNetwork/project-id-global-vpc1-vpc # kpt-set: compute.cnrm.cloud.google.com/namespaces/networking/ComputeNetwork/${project-id}-global-vpc1-vpc
+spec:
+  resourceID: nane1-vpc1-apprz-snet # kpt-set: nane1-vpc1-apprz-snet
+  ipCidrRange: 10.1.2.0/24
+  region: northamerica-northeast1
+  description: northamerica-northeast1 APPRZ subnet
+  privateIpGoogleAccess: true
+  networkRef:
+    name: project-id-global-vpc1-vpc # kpt-set: ${project-id}-global-vpc1-vpc
+  logConfig:
+    aggregationInterval: INTERVAL_5_SEC
+    flowSampling: 0.5
+    metadata: INCLUDE_ALL_METADATA
+---
+# Subnet DATARZ northamerica-northeast1
+# SC-7 BOUNDARY PROTECTION
+apiVersion: compute.cnrm.cloud.google.com/v1beta1
+kind: ComputeSubnetwork
+metadata:
+  name: project-id-nane1-vpc1-datarz-snet # kpt-set: ${project-id}-nane1-vpc1-datarz-snet
+  namespace: networking
+  annotations:
+    cnrm.cloud.google.com/project-id: project-id # kpt-set: ${project-id}
+    config.kubernetes.io/depends-on: compute.cnrm.cloud.google.com/namespaces/networking/ComputeNetwork/project-id-global-vpc1-vpc # kpt-set: compute.cnrm.cloud.google.com/namespaces/networking/ComputeNetwork/${project-id}-global-vpc1-vpc
+spec:
+  resourceID: nane1-vpc1-datarz-snet # kpt-set: nane1-vpc1-datarz-snet
+  ipCidrRange: 10.1.3.0/24
+  region: northamerica-northeast1
+  description: northamerica-northeast1 DATARZ subnet
+  privateIpGoogleAccess: true
+  networkRef:
+    name: project-id-global-vpc1-vpc # kpt-set: ${project-id}-global-vpc1-vpc
+  logConfig:
+    aggregationInterval: INTERVAL_5_SEC
+    flowSampling: 0.5
+    metadata: INCLUDE_ALL_METADATA
+---
+# Subnet PAZ northamerica-northeast2
+# SC-7 BOUNDARY PROTECTION
+apiVersion: compute.cnrm.cloud.google.com/v1beta1
+kind: ComputeSubnetwork
+metadata:
+  name: project-id-nane2-vpc1-paz-snet # kpt-set: ${project-id}-nane2-vpc1-paz-snet
+  namespace: networking
+  annotations:
+    cnrm.cloud.google.com/project-id: project-id # kpt-set: ${project-id}
+    config.kubernetes.io/depends-on: compute.cnrm.cloud.google.com/namespaces/networking/ComputeNetwork/project-id-global-vpc1-vpc # kpt-set: compute.cnrm.cloud.google.com/namespaces/networking/ComputeNetwork/${project-id}-global-vpc1-vpc
+spec:
+  resourceID: nane2-vpc1-paz-snet # kpt-set: nane2-vpc1-paz-snet
```

Comment:
Maybe also missing ${project-id}?

Reply:
you are right luc, I thought it was a comment for the other line. When resource id is not specified, the resource name in gcp will have the same value as the metadata name. I'm setting a very complex metadata name to ensure uniqueness accross the k8s namespace but the resource name only need to be unique within it's project so I'm defining a much simpler resource name. As you pointed out, I will remove the kpt-set because it is not required.